pkg/client/aws/strategies_aws.go

```diff
@@ -177,89 +194,43 @@ func reconcileRedisStrategy(config, tier, backupTimeStart, maintenanceTimeStart
 	return string(marshalledStrategy), nil
 }
 
-// builder function to provide default data for aws strategy config map
-func buildDefaultAWSConfigMap() map[string]string {
-	return map[string]string{
-		blobstorageStratKey: buildDefaultAWSStratValue(),
-		redisStratKey:       buildDefaultAWSStratValue(),
-		postgresStratKey:    buildDefaultAWSStratValue(),
-	}
-}
-
-// builder function to provide default value used per tier in aws strategy config map
-func buildDefaultAWSStratValue() string {
-	return "{\"development\": { \"region\": \"\", \"createStrategy\": {}, \"deleteStrategy\": {} }, \"production\": { \"region\": \"\", \"createStrategy\": {}, \"deleteStrategy\": {} }}"
-}
-
 // build aws maintenance and backup windows
-func buildAWSWindows(timeConfig *StrategyTimeConfig) (string, string, error) {
-	// ensure backup applyOn format is correct
-	parsedBackupTime, err := time.Parse("15:04", timeConfig.BackupStartTime)
-	if err != nil {
-		return "", "", errorUtil.Wrapf(err, "failed to parse backup ApplyOn value")
-	}
-
-	// correct format is assumed by the time this function is called
-	maintenanceWindowSegments := strings.Split(timeConfig.MaintenanceStartTime, " ")
-	parsedMaintenanceTime, err := time.Parse("15:04", maintenanceWindowSegments[1])
-	if err != nil {
-		return "", "", errorUtil.Wrapf(err, "failed to parse maintenance ApplyFrom value")
-	}
-
-	// ensure ddd is correct and overlaps to the following day if block starts after 23 hours
-	// set window day from value from segmented string
-	maintenanceWindowDayFrom := strings.ToLower(maintenanceWindowSegments[0][:3])
-	// map of valid day with time.weekday int value
-	var validMaintenanceDays = map[string]int{
-		"sun": 0,
-		"mon": 1,
-		"tue": 2,
-		"wed": 3,
-		"thu": 4,
-		"fri": 5,
-		"sat": 6,
-	}
-
-	// get current maintenance dayTo value from maintenance dayFrom value
-	var maintenanceTimeTo int
-	for day, value := range validMaintenanceDays {
-		if maintenanceWindowDayFrom == day {
-			maintenanceTimeTo = value
-		}
-	}
-
+func buildAWSWindows(timeConfig *stratType.StrategyTimeConfig) (string, string, error) {
```

Comment:
The idea here was that the instead of having to duplicate the complex logic for parsing an arbitrary string format the configuration could be represented by a machine readable data structure instead.